### ISSUE TITLE
Adjusting the size of a stopped thread pool starts new threads.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7011]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)| @tomprince reported|
|-|-|
|Trac ID|trac#7011|
|Type|defect|
|Created|2014-03-01 00:09:08Z|
|Branch|https://github.com/twisted/twisted/tree/threadpool-adjust-hang-7011|

The following code will hang:
```python
from twisted.python.threadpool import ThreadPool
pool = ThreadPool()
pool.start()
pool.stop()
pool.adjustPoolsize()
```

<details><summary>Searchable metadata</summary>

```
trac-id__7011 7011
type__defect defect
reporter__tom_prince tom.prince
priority__normal normal
milestone__ 
branch__branches_threadpool_adjust_hang_7011 branches/threadpool-adjust-hang-7011
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1393632548000000 1393632548000000
changetime__1393681582000000 1393681582000000
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [41781]) Branching to threadpool-adjust-hang-7011.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

The pool didn't mark itself as not started when it was stopped - I think this is an oversight? Unless it means it was never started... anyway, I've added that and a test for that case. It hangs as tomprince says if I run it on master.

Putting up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @hawkowl|
|-|-|

Thanks.  Just a few minor comments:

  1. _Making sure that a stopped pool doesn't start new workers when the size is adjusted._ isn't a proper test method docstring.
  1. The `ThreadPool.threads` attribute is undocumented but the new test relies on it having a particular meaning.  Please document that meaning.  I think the expected behavior is already tested by `test_start` but if you wanted to add a more direct test that would be cool too.
  1. `ThreadPool.started` is also undocumented.  Can you document it?
  1. In the news fragment I would say "but the pool is stopped" rather than "and the pool is stopped".  Alternatively, "and the pool is not running."  "stopped" could indicate a state or an action.

Please merge after addressing those points.  Thanks again.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [41786]) Merge threadpool-adjust-hang-7011

Author: hawkowl
Reviewer: exarkun
Fixes: [#7011](https://github.com/twisted/twisted/issues/7011)

twisted.python.threadpool.ThreadPool no longer starts new workers when its pool size is changed while the pool is not running.
